h2o-core/src/main/java/water/Value.java

```diff
@@ -498,7 +512,7 @@ void lowerActiveGetCount( H2ONode h2o ) {
       int old = _rwlock.get(); // Read the lock-word
       assert old > 0;      // Since lowering, must be at least 1
       assert old != -1;    // Not write-locked, because we are an active reader
-      assert (h2o==null) || (_replicas!=null && _replicas[h2o._unique_idx]==1); // Self-bit is set
+      assert (h2o==null) || (_replicas!=null && (h2o._unique_idx >= _replicas.length || _replicas[h2o._unique_idx]==1)); // Self-bit is set
```

Comment:
I'm having hard time understanding this change here `(h2o._unique_idx >= _replicas.length` cold you please explain it a little?

Reply:
For "too many" nodes we stop registering replicas. We just invalidate every key if the node ever touched DKV.